girder/utility/gridfs_assetstore_adapter.py

```diff
@@ -79,31 +84,45 @@ def validateInfo(doc):
 
     @staticmethod
     def fileIndexFields():
-        return ['sha512']
+        return ['sha512', 'chunkUuid']
 
     def __init__(self, assetstore):
         """
         :param assetstore: The assetstore to act on.
         """
         super(GridFsAssetstoreAdapter, self).__init__(assetstore)
+        recent = False
+        connectionArgs = (self.assetstore.get('mongohost', None),
+                          self.assetstore.get('replicaset', None))
+        try:
+            # Guard in case the connectionArgs is unhashable
+            key = connectionArgs
+            if key in _recentConnections:
+                recent = (time.time() - _recentConnections[key]['created'] <
+                          RECENT_CONNECTION_CACHE_TIME)
+        except Exception:
+            key = None
         try:
-            self.chunkColl = getDbConnection(
-                self.assetstore.get('mongohost', None),
-                self.assetstore.get('replicaset', None)
-            )[self.assetstore['db']].chunk
-            _ensureChunkIndices(self.chunkColl)
+            # MongoClient automatically reuses connections from a pool, however
+            # getting the connection takes time, so use our own cache.  We also
```

Comment:
> however getting the connection takes time, so use our own cache

It doesn't look like we actually cache the connections themselves, is this accurate?